OpenSim/Common/Component.h

```diff
@@ -712,6 +716,7 @@ OpenSim_DECLARE_ABSTRACT_OBJECT(Component, Object);
     */
     template <class C = Component>
     C& updComponent(const std::string& name) {
+        clearObjectIsUpToDateWithProperties();
```

Comment:
> if you access bodies, forces, constraint via their Sets (e.g, `Model::updBodySet()`) these have the same effect of *dirtying* the up-to-date flag.

Ah, ok. I didn't know that.
![](https://media.giphy.com/media/83QtfwKWdmSEo/giphy.gif)

A separate potential issue:
IIRC `get...()` and `upd...()` can both work in scripting. Is it then possible for an assumption about the up-to-date flag to be incorrect in scripting?